kt/entry-generation/godot-class-graph-symbol-processor/src/main/kotlin/godot/annotation/processor/classgraph/extensions/AnnotationExtensions.kt

```diff
@@ -0,0 +1,181 @@
+import godot.annotation.RegisterProperty
+import godot.annotation.RegisterSignal
+import godot.annotation.Rpc
+import godot.annotation.RpcMode
+import godot.annotation.Sync
+import godot.annotation.Tool
+import godot.annotation.TransferMode
+import godot.annotation.processor.classgraph.Context
+import godot.annotation.processor.classgraph.ErrorsDatabase
+import godot.annotation.processor.classgraph.constants.SET
+import godot.entrygenerator.model.ColorNoAlphaHintAnnotation
+import godot.entrygenerator.model.DirHintAnnotation
+import godot.entrygenerator.model.EnumFlagHintStringAnnotation
+import godot.entrygenerator.model.ExpEasingHintAnnotation
+import godot.entrygenerator.model.ExportAnnotation
+import godot.entrygenerator.model.FileHintAnnotation
+import godot.entrygenerator.model.GodotAnnotation
+import godot.entrygenerator.model.GodotBaseTypeAnnotation
+import godot.entrygenerator.model.IntFlagHintAnnotation
+import godot.entrygenerator.model.MultilineTextHintAnnotation
+import godot.entrygenerator.model.PlaceHolderTextHintAnnotation
+import godot.entrygenerator.model.Range
+import godot.entrygenerator.model.RangeHintAnnotation
+import godot.entrygenerator.model.RegisterClassAnnotation
+import godot.entrygenerator.model.RegisterFunctionAnnotation
+import godot.entrygenerator.model.RegisterPropertyAnnotation
+import godot.entrygenerator.model.RegisterSignalAnnotation
+import godot.entrygenerator.model.RpcAnnotation
+import godot.entrygenerator.model.ToolAnnotation
+import io.github.classgraph.AnnotationEnumValue
+import io.github.classgraph.AnnotationInfo
+import io.github.classgraph.ClassRefTypeSignature
+import io.github.classgraph.FieldInfo
+
+fun AnnotationInfo.mapToGodotAnnotation(parentDeclaration: Any): GodotAnnotation? {
+    @Suppress("UNCHECKED_CAST")
+    return when (name) {
+        RegisterClass::class.java.name -> RegisterClassAnnotation(
+            customName = parameterValues.getValue("customName") as? String,
+            symbolProcessorSource = this
+        )
+        RegisterFunction::class.java.name -> RegisterFunctionAnnotation(this)
+        RegisterProperty::class.java.name -> RegisterPropertyAnnotation(this)
+        RegisterSignal::class.java.name -> RegisterSignalAnnotation(this)
+        Tool::class.java.name -> ToolAnnotation(this)
+        Export::class.java.name -> ExportAnnotation(this)
+        Rpc::class.java.name -> RpcAnnotation(
+            rpcMode = getRpcMode(),
+            sync = getSyncMode(),
+            transferMode = getTransferMode(),
+            transferChannel = parameterValues.getValue("transferChannel") as Int,
+            symbolProcessorSource = this
+        )
+        "godot.annotation.GodotBaseType" -> GodotBaseTypeAnnotation(this) // is internal
+        EnumFlag::class.java.name -> {
+            if (parentDeclaration !is FieldInfo) {
+                ErrorsDatabase.add(
+                    "EnumFlag annotation should be placed on property."
```

Comment:
I'd add where this error occurred. If you have multiple such error's then it's hard to find. Maybe add the declaration to the error db as well and assemble the final error message when you print them?

Reply:
There are similar cases at other places